clash-verge-rev: 1.6.6 -> 1.7.5

Guanran928 opened the pull request:
Diff: https://github.com/clash-verge-rev/clash-verge-rev/compare/v1.6.6...v1.7.5

## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Guanran928 committed:
clash-verge-rev: 1.6.6 -> 1.7.2

Diff: https://github.com/clash-verge-rev/clash-verge-rev/compare/v1.6.6...v1.7.2

Guanran928 committed:
clash-verge-rev: 1.7.2 -> 1.7.3

Diff: https://github.com/clash-verge-rev/clash-verge-rev/compare/v1.7.2...v1.7.3

Bot-wxt1221 commented:
As I know, clash verge doesn't provide any libraries to other programs. And It just release on the main branch.

We should have the newest version for 24.05 as well I think.

Guanran928 commented:
> As I know, clash verge doesn't provide any libraries to other programs. And It just release on the main branch.
> 
> We should have the newest version for 24.05 as well I think.

1.7.0 introduced breaking changes.

https://github.com/clash-verge-rev/clash-verge-rev/releases/tag/v1.7.0

Guanran928 committed:
clash-verge-rev: 1.7.3 -> 1.7.5

Diff: https://github.com/clash-verge-rev/clash-verge-rev/compare/v1.7.3...v1.7.5

SuperSandro2000 commented on pkgs/by-name/cl/clash-verge-rev/package.nix:
```diff
@@ -1,19 +1,67 @@
+  buildInputs = [
+    openssl
+    webkitgtk
+    stdenv.cc.cc
```
Isn't cc.cc included by default in the stdenv?

Guanran928 replied:
I copied them from clash-verge. https://github.com/NixOS/nixpkgs/blob/aeda87464f545fbe4ebb26a79e547b097b04789a/pkgs/by-name/cl/clash-verge/package.nix#L33

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/4281

Guanran928 commented:
Rebased on latest master as an attempt to fix red CI.
